### ISSUE TITLE
Rule request - Source file should have BLOCK-LEVEL ON ERROR UNDO, THROW

### ISSUE THREAD

**Issue by movedoa** [Riverside-Software/sonar-openedge#1071]
Our application started a long time before catch was a thing in the abl so nearly all of the code is written und the assumption of "classic" error handling, meaning leaving the block, raise the error, keep running.

Since we want to be able to catch unexpected errors we want to migrate everything to BLOCK-LEVEL ON ERROR UNDO, THROW but this takes time. A rule that marks files that dont have the statement would be very useful.
It would allow us to do a steady migration file by file without missing anything and then finally use -undothrow 2.

**Comment by kenherring-rkt** [comment]
We accomplished this with some simple linting.  We actually only want the `block-level...` statement in out unit tests, but since tests don't really show results it wouldn't be useful as a sonar rule.  We use some similar to this to fail if any files are found:

```bash
if git grep -L -r "block-level" . | grep '/unit/*Test.cls'; then
  echo "ERROR: found tests without 'BLOCK-LEVEL' statement."
  exit 1
fi
```

* `git grep -L "block-level"` - returns files without any `block-level` text found
* `grep `/unit/*Test.cls` - limits the results to only the test cases we're interested in.

**Comment by kenherring-rkt** [comment]
That said, I'd be interested in using a rule if it existed as well, but only if it was reporting for unit tests (as defined by `sonar.tests` and `sonar.test.inclusions`).  I think this goes against sonar use cases though if I understand those parameters correctly.

**Comment by gquerret** [comment]
Did you have a look at the [No error handling directive](https://sonar.riverside-software.fr/coding_rules?languages=oe&q=error&open=rssw-oe-main%3Aeu.rssw.antlr.proparse.checks.ErrorHandlingDirective) rule?

**Comment by movedoa** [comment]
> Did you have a look at the [No error handling directive](https://sonar.riverside-software.fr/coding_rules?languages=oe&q=error&open=rssw-oe-main%3Aeu.rssw.antlr.proparse.checks.ErrorHandlingDirective) rule?

Didn't find that somehow. Looks like this should do the trick.
